### PR TITLE
Working

### PR DESCRIPTION
**Pull request by jmrosinski**

### CODE REVIEW

File: test/Makefile.am [hunk 1]
```diff
@@ -7,61 +7,76 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 
 # Link to GPTL fortran library.
-AM_LDFLAGS = ${top_builddir}/src/libgptlf.la
+AM_LDFLAGS = ${top_builddir}/src/libgptlf.la -rdynamic
```
Comment (edhartnett) [inline test/Makefile.am]: Do we have to set -rdynamic? Is it guaranteed to be supported on all compilers?

File: test/Makefile.am [hunk 2]
```diff
@@ -7,61 +7,76 @@
...
 # For combined C/Fortran builds, find the C library.
+# Be careful making changes here: If -lunwind is needed it must come at the end of link cmds
 if BUILD_PACKAGE
 LDADD = ${top_builddir}/../GPTL/src/libgptl.la
+else
+LDADD =
+endif
+
+if HAVE_LIBUNWIND
+LDADD += -lunwind
```
Comment (edhartnett) [inline test/Makefile.am]: Where does libunwind come from?

File: test/Makefile.am [hunk 3]
```diff
@@ -7,61 +7,76 @@
...
-pmpi_SOURCES = pmpi.F90 ${MODFILE}
+TESTS          += pmpi
+pmpi_SOURCES    = pmpi.F90 ${MODFILE}
+endif
 
-check_PROGRAMS += threadohd
-#TESTS += threadohd
-threadohd_SOURCES = threadohd.F90 ${MODFILE}
+check_PROGRAMS += summary
+TESTS          += summary
```
Comment (edhartnett) [inline test/Makefile.am]: If you are manually editing the spaces in here you should probably stop. I use emacs and just hit the reindent freely, which gives a minimalist standard spacing, and will overwrite any manual spacing.

**Comment by jmrosinski** [comment]
-rdynamic is required for any program to be auto-profiled. It works with
GNU and Intel. I should try PGI. Don't know about Mac, but the OpenMP
issues on Mac are show-stoppers.

libunwind is the supposed newest/best way to get the compiler to unwind the
current stack and return function names. There's also backtrace, which only
requires -rdynamic not -lunwind, but doesn't work as well. So I added
support for both in GPTL. This is much of what I've been workin on the past
few weeks--to get auto-profiling to work with shared objects.

**Comment by jmrosinski** [comment]
Oh, forgot to mention, libunwind and/or -rdynamic are only needed for codes
which are being auto-profiled. I couldn't figure out the Makefile.am
structure enough to know how to just apply to selected files, so hit it
with too big a hammer.

**Comment by edhartnett** [comment]
So libunwind is a library that must exist on the system? It should be checked in configure.ac.

Once checked in configure.ac, you don't need to add it to the library flags in the makefiles, that will happen automatically. You will add an AC_CHECK_LIB macro to configure.ac. Look at what's there now for examples. If the library can't be found, you can error out in configure, or do whatever else seems best.

Since this is not passing I would suggest starting a new branch, and putting up a subset of these changes, breaking the problem up.

Go ahead with the -rdynamic flag for now. We will circle back to that after we get these changes through travis. (But perhaps this is what is breaking travis?).

**Comment by edhartnett** [comment]
When you say "auto-profiled" you mean the HAVE_BACKTRACE macro is true? (See configure.ac).

Libraries should be found in configure.ac, not in Makefile.am.

**Comment by jmrosinski** [comment]
Yes I added checks for it in configure.ac. Please have a look (in
"working") and see if I did it correctly. Note the comment about
"weirdness" in libunwind which forced me to use a header checker rather
than AC_CHECK_LIB.

**Comment by jmrosinski** [comment]
Auto-profiled means *either *HAVE_BACKTRACE *or *HAVE_LIBUNWIND are set. I
think I added the right stuff to configure.ac to ensure that only one of
these can be set, and HAVE_BACKTRACE is the default. If the user is going
to auto-profile code, they also need to add -finstrument-functions to the
*FLAGS of the file(s) containing the routine(s) they wish to profile.

**Comment by edhartnett** [comment]
OK, so this needs to be figured out in configure.ac, and the flag set if needed, rather than always setting it in Makefile.am

**Pull request merged**